controllers/istiobase/istiobase_controller.go

```diff
@@ -0,0 +1,244 @@
+	log.Info("Uninstalling Helm chart")
+	_, err = r.ChartManager.UninstallChart(ctx, baseReleaseName, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to uninstall Helm chart %q: %w", baseChartName, err)
+	}
+	return nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	logger := mgr.GetLogger().WithName("ctrlr").WithName("istiobase")
+
+	enqueueSingleton := handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+		return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: "default"}}}
+	})
+
+	eventHandler := wrapEventHandler(logger, enqueueSingleton)
+
+	managedByPredicate := predicate.NewPredicateFuncs(func(obj client.Object) bool {
+		return obj.GetLabels()[constants.ManagedByLabelKey] == constants.ManagedByLabelValue
+	})
+
+	return ctrl.NewControllerManagedBy(mgr).
+		WithOptions(controller.Options{
+			LogConstructor: func(_ *reconcile.Request) logr.Logger {
+				log := logger
+				return log
+			},
+		}).
+		Named("istiobase").
+
+		// We use predicate.IgnoreUpdate() so that we skip the reconciliation when a pull secret is added to the ServiceAccount.
+		// This is necessary so that we don't remove the newly-added secret.
+		// TODO: this is a temporary hack until we implement the correct solution on the Helm-render side
+		Watches(&corev1.ServiceAccount{}, eventHandler, builder.WithPredicates(predicate2.IgnoreUpdate(), managedByPredicate)).
+		Watches(&admissionv1.ValidatingWebhookConfiguration{}, eventHandler, builder.WithPredicates(validatingWebhookConfigPredicate(), managedByPredicate)).
+		Watches(&admissionv1.ValidatingAdmissionPolicy{}, eventHandler, builder.WithPredicates(managedByPredicate)).
+		Watches(&admissionv1.ValidatingAdmissionPolicyBinding{}, eventHandler, builder.WithPredicates(managedByPredicate)).
+		Watches(&v1.IstioRevision{}, eventHandler).
+		Watches(&v1.IstioRevisionTag{}, eventHandler).
+		Complete(reconcile.Func(r.Reconcile))
+}
+
+func wrapEventHandler(logger logr.Logger, handler handler.EventHandler) handler.EventHandler {
+	return enqueuelogger.WrapIfNecessary("IstioBase", logger, handler)
+}
+
+func validatingWebhookConfigPredicate() predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.TypedUpdateEvent[client.Object]) bool {
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				return false
+			}
+
+			if e.ObjectNew.GetName() == webhookName {
+				// Istiod updates the caBundle and failurePolicy fields. We must ignore changes to these fields to prevent an endless update loop.
+				clearIgnoredFields(e.ObjectOld)
+				clearIgnoredFields(e.ObjectNew)
+				return !reflect.DeepEqual(e.ObjectNew, e.ObjectOld)
+			}
+			return true
+		},
+	}
+}
+
+func clearIgnoredFields(obj client.Object) {
+	obj.SetResourceVersion("")
+	obj.SetGeneration(0)
+	obj.SetManagedFields(nil)
```

Comment:
Do any of the controllers use SSA? If not, we can probably set [this](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.20.4/pkg/cache#TransformStripManagedFields) on the manager/cache.

Reply:
This has to be done globally, when instantiating the manager, right? So, best to do it in a separate PR.